packages/jsActions/nanoflow-actions-native/src/geolocation/RequestLocationPermission.ts

```diff
@@ -14,11 +15,78 @@ import Geolocation, { GeolocationStatic } from "@react-native-community/geolocat
 export async function RequestLocationPermission(): Promise<boolean> {
     // BEGIN USER CODE
 
-    if (navigator && navigator.product === "ReactNative") {
-        if (!navigator.geolocation) {
-            (navigator.geolocation as GeolocationStatic) = Geolocation;
+    const hasPermissionIOS = async (): Promise<boolean> => {
+        const openSetting = (): void => {
+            Linking.openSettings().catch(() => {
+                Alert.alert("Unable to open settings.");
+            });
+        };
+
+        return (navigator.geolocation as any)
+            .requestAuthorization("whenInUse")
+            .then((status: "granted" | "denied" | "disabled") => {
+                if (status === "granted") {
+                    return true;
+                }
+
+                if (status === "denied") {
+                    Alert.alert("Location permission denied.");
+                }
+
+                if (status === "disabled") {
+                    Alert.alert("Turn on Location Services to allow to determine your location.", "", [
+                        { text: "Go to Settings", onPress: openSetting },
+                        {
+                            text: "Don't Use Location"
+                        }
+                    ]);
+                }
+
+                return false;
+            });
+    };
+
+    const hasPermissionAndroid = async (): Promise<boolean> => {
+        if (Platform.OS === "android" && Platform.Version < 23) {
```

Comment:
is this first check necessary since the caller checks which platform it is before delegating to hasPermissionIOS or hasPermissionsAndroid?